content/blog/base-vs-magrittr-pipe/index.Rmd

```diff
@@ -0,0 +1,68 @@
+-->
+
+**Note:** The following has been adapted from a section of the forthcoming second edition of [R for Data Science](https://r4ds.hadley.nz/) that had to be removed due to length limitations.
+
+## Pipes
+
+With the 4.1.0 release of R, a native pipe operator, `|>`, was introduced.
+Its behavior is by and large the same as that of the [`%>%`](https://magrittr.tidyverse.org/reference/pipe.html) pipe provided by the **magrittr** package. 
+Both operators (`|>` and `%>%`) let you "pipe" an object forward to a function or call expression, thereby allowing you to express a sequence of operations that transform an object.
+
+To learn more about the basic utility of pipes, see [The pipe](https://r4ds.hadley.nz/data-transform.html#the-pipe) section of R for Data Science.
+
+## `|>` vs. `%>%`
+
+While `|>` and `%>%` behave identically for simple cases, there are a few crucial differences.
+These are most likely to affect you if you're a long-term user of `%>%` who has taken advantage of some of the more advanced features.
+But they're still good to know about even if you've never used `%>%` because you're likely to encounter some of them when reading wild-caught code.
+
+-   By default, the pipe passes the object on its left-hand side to the first argument of the function on the right-hand side.
+    `%>%` allows you to change the placement with a `.` placeholder.
+    For example, `x %>% f(1)` is equivalent to `f(x, 1)` but `x %>% f(1, .)` is equivalent to `f(1, x)`.
+    R 4.2.0 added a `_` placeholder to the base pipe, with one additional restriction: the argument has to be named.
+    For example, `x |> f(1, y = _)` is equivalent to `f(1, y = x)`.
+
+-   The `|>` placeholder is deliberately simple and can't replicate many features of the `%>%` placeholder: you can't pass it to multiple arguments, and it doesn't have any special behavior when the placeholder is used inside another function.
+    For example, `df %>% split(.$var)` is equivalent to `split(df, df$var)`, and `df %>% {split(.$x, .$y)}` is equivalent to `split(df$x, df$y)`.
+
+    With `%>%`, you can use `.` on the left-hand side of operators like `$`, `[[`, `[` , so you can extract a single column from a data frame with (e.g.) `mtcars %>% .$cyl`.
+    A future version of R may add similar support for `|>` and `_`.
+    For the special case of extracting a column out of a data frame, you can also use `dplyr::pull()`:
+
+    ```{r eval=FALSE}
+    mtcars |> pull(cyl)
+    ```
+
+-   `%>%` allows you to drop the parentheses when calling a function with no other arguments; `|>` always requires the parentheses.
+
+-   `%>%` allows you to start a pipe with `.` to create a function rather than immediately executing the pipe; this is not supported by the base pipe.
+
+Luckily there's no need to commit entirely to one pipe or the other --- you can use the base pipe for the majority of cases where it's sufficient and use the magrittr pipe when you really need its special features.
```

Comment:
I feel like this is going to be the main take home for most people - that you can use base pipe most of the time, and magrittr in special cases. 
This is a short enough post that this sentence should stand out being at the end, but another option could be to move this to just before the ## `|>` vs. `%>%` header.